app/vmalert/rule/group.go

```diff
@@ -724,13 +725,19 @@ func (e *executor) exec(ctx context.Context, r Rule, ts time.Time, resolveDurati
 	return errGr.Err()
 }
 
+var bbPool bytesutil.ByteBufferPool
+
 // getStaleSeries checks whether there are stale series from previously sent ones.
 func (e *executor) getStaleSeries(r Rule, tss []prompbmarshal.TimeSeries, timestamp time.Time) []prompbmarshal.TimeSeries {
+	bb := bbPool.Get()
+	defer bbPool.Put(bb)
+
 	ruleLabels := make(map[string][]prompbmarshal.Label, len(tss))
 	for _, ts := range tss {
-		// convert labels to strings so we can compare with previously sent series
-		key := labelsToString(ts.Labels)
-		ruleLabels[key] = ts.Labels
+		// convert labels to strings, so we can compare with previously sent series
+		bb.B = labelsToString(bb.B, ts.Labels)
+		ruleLabels[string(bb.B)] = ts.Labels
```

Comment:
Since ts.Labels is used as key in map ruleLabels, the value can be nil. And add `stringToLabels(s string) []prompbmarshal.Label` to get original labels when needed.
```
	ruleLabels := make(map[string]interface{}, len(tss))
```
It should decrease memory usage more, wdyt @hagen1778 
```
BenchmarkGetStaleSeries
BenchmarkGetStaleSeries-10    	   55296	     19232 ns/op	   14363 B/op	     103 allocs/op
```